wipac_dev_tools/setup_builder.py

```diff
@@ -0,0 +1,344 @@
+"""Module to build `setup.cfg` sections for use by `setup.py`/`setuptools`.
+
+Used in CI/CD, used by GH Action.
+"""
+
+import argparse
+import configparser
+import os
+import re
+from dataclasses import dataclass
+from typing import Iterator, List, Tuple, cast
+
+import requests
+
+BUIDLER_SECTION_NAME = "wipac:cicd_setup_builder"
+GENERATED_STR = f"generated by {BUIDLER_SECTION_NAME}"
+AUTHOR = "WIPAC Developers"
+AUTHOR_EMAIL = "developers@icecube.wisc.edu"
+DEFAULT_KEYWORDS = ["WIPAC", "IceCube"]
+LICENSE = "MIT"
+
+PythonMinMax = Tuple[Tuple[int, int], Tuple[int, int]]
+
+
+def get_latest_py3_release() -> Tuple[int, int]:
+    """Return the latest python3 release version as tuple."""
+    minor = 10  # start with 3.10
+    while True:
+        url = f"https://docs.python.org/release/3.{minor}.0/"
+        if requests.get(url).status_code == 404:
```

Comment:
May want to use `status_code >= 300` in case redirects (3xx) or server errors (5xx) show up; might get stuck in an endless loop if we're looking for a 404 that never comes.

May want to add a cap at minor = 50 and bail out with an error if you reach it.

Reply:
Good idea. It'll probably never happen, but this fix is a safer bet.